src/algorithms/rhythm/tempocnn.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2006-2020  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#ifndef ESSENTIA_TEMPOCNN_H
+#define ESSENTIA_TEMPOCNN_H
+
+#include "algorithmfactory.h"
+#include "essentiamath.h"
+
+namespace essentia {
+namespace standard {
+
+class TempoCNN : public Algorithm {
+
+ protected:
+  Input<std::vector<Real> > _audio;
+  Output<Real> _globalTempo;
+  Output<std::vector<Real> > _localTempo;
+  Output<std::vector<Real> > _localTempoProbs;
+
+  Algorithm* _tensorflowPredictTempoCNN;
+
+  std::string _aggregationMethod;
+
+  std::vector<std::vector<Real> > _predictions;
+
+ public:
+  TempoCNN() {
+    declareInput(_audio, "audio", "the input audio signal sampled at 11025 Hz");
+    declareOutput(_globalTempo, "globalTempo" , "the overall tempo estimation in BPM");
+    declareOutput(_localTempo, "localTempo", "the patch-wise tempo estimations in BPM");
+    declareOutput(_localTempoProbs, "localTempoProbabilities", "the patch-wise tempo probabilities");
+
+    _tensorflowPredictTempoCNN = AlgorithmFactory::create("TensorflowPredictTempoCNN");
+  }
+
+  ~TempoCNN() {
+    if (_tensorflowPredictTempoCNN) delete _tensorflowPredictTempoCNN;
+  }
+
+  void declareParameters() {
+    declareParameter("graphFilename", "the name of the file containing the model to use", "", Parameter::STRING);
+    declareParameter("input", "the name of the input node in the TensorFlow graph", "", "input");
+    declareParameter("output", "the name of the node from which to retrieve the output tensors", "", "output");
```

Comment:
Make sure if the behavior is correct if a user specifies different `output`. As I understand, the intention here is to leave the input / output nodes flexible and up to the user's responsibility.  Should we leave it like that, add a line in the DOC with a note, or have an E_WARNING to make sure the user knows what they do?

Reply:
In the more general `TensorflowPredict`* algorithms `output` is useful to:
1. Match the graph's output node name
2. Retrieve any other node from the graph

While  2) doesn't make sense in `TempoCNN`, 1) is still a valid reason.

For example, different `TempoCNN` models have different output node names in the original implementation (e.g., `output_46`, `output_47`...). Despite we standardize the input and output node names when we serialize the models, `output` would still be useful for a user that serializes the models by himself/herself without renaming the nodes following our convention.

Reply:
I'll rename the parameter →  "the name of the node from which to retrieve the tempo bins activations"

Reply:
Yes, this sounds more clear 👍